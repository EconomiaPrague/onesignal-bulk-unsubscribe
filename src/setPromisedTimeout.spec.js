const setPromisedTimeout = require('./setPromisedTimeout')

describe('setPromisedTimeout', () => {
  const testValue = 'test value'

  beforeAll(() => {
    jest.useFakeTimers()
  })

  test('should return a Promise', () => {
    const returnValue = setPromisedTimeout(() => {})

    expect(returnValue).toBeInstanceOf(Promise)
  })

  test("should resolve with provided function' return value", async () => {
    const returnValue = setPromisedTimeout(() => testValue)
    jest.runAllTimers()
    return expect(returnValue).resolves.toBe(testValue)
  })

  test("should resolve with provided function' resolved value", async () => {
    const asyncValue = setPromisedTimeout(async () => testValue)
    jest.runAllTimers()
    return expect(asyncValue).resolves.toBe(testValue)
  })

  test('should reject when provided function does', async () => {
    const asyncValue = setPromisedTimeout(async () => {
      throw new Error('test error')
    })
    jest.runAllTimers()
    return expect(asyncValue).rejects.toThrowErrorMatchingSnapshot()
  })
})
