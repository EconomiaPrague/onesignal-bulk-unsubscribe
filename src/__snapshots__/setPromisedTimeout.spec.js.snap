// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`setPromisedTimeout should reject when provided function does 1`] = `"test error"`;
